core/templates/dev/head/pages/exploration_editor/simple_editor_tab/questions/MultipleChoiceEditorDirective.js

```diff
@@ -136,21 +136,30 @@ oppia.directive('multipleChoiceEditor', [
           $scope.saveChoice = function(index, newChoiceValue) {
             if (!newChoiceValue) {
               alertsService.addWarning('Cannot save an empty choice.');
-              return;
+              return {
```

Comment:
As far as caller code is concerned, we can add documentations for the specific method which is throwing error.

But as you said, if the error is recoverable then we should return `status` object and update the documentation accordingly.